### PR TITLE
Fcm 

### PR DESCRIPTION
**Pull request by Ekwav**
Token creation

### CODE REVIEW

File: public/preScript.js [hunk 1]
```diff
@@ -3,7 +3,7 @@ window.websocket.onopen = function() {
     console.log("Websocket opened");
 }
 
-if (window.location.href.toString().indexOf("localhost") !== -1) {
+if (window.location.href.toString().indexOf("localhost") !== -1 && false) {
```
Comment (matthias-luger) [inline public/preScript.js:6]: remove && false

File: public/preScript.js [hunk 2]
```diff
@@ -13,6 +13,12 @@ if (window.location.href.toString().indexOf("localhost") !== -1) {
             console.log('ServiceWorker registration successful with scope: ', registration.scope);
 
             loadScript("https://arc.io/widget.min.js#WK8BPDas");
+
```
Comment (matthias-luger) [inline public/preScript.js:16]: remove empty lines

File: public/preScript.js [hunk 3]
```diff
@@ -13,6 +13,12 @@ if (window.location.href.toString().indexOf("localhost") !== -1) {
...
+
+            loadScript('https://www.gstatic.com/firebasejs/8.2.2/firebase-app.js');
+            loadScript('https://www.gstatic.com/firebasejs/8.2.2/firebase-messaging.js');
+
+            setTimeout(() => pushNotifications(registration), 3000);
```
Comment (matthias-luger) [inline public/preScript.js:21]: setTimeout, really necessary????

File: public/preScript.js [hunk 4]
```diff
@@ -29,4 +35,31 @@ function loadScript(url) {
     script.async = true;
     script.src = url;
     document.getElementsByTagName("head")[0].appendChild(script);
+}
+
+async function pushNotifications(serviceWorkerRegistration) {
```
Comment (matthias-luger) [inline public/preScript.js:40]: no async await

File: src/routes.tsx [hunk 5]
```diff
@@ -27,6 +27,9 @@ function isTrackingAllowed() {
   return false;
 }
 
+function initFCM() {
```
Comment (matthias-luger) [inline src/routes.tsx:30]: remove unused function

File: src/api/ApiTypes.d.tsx [hunk 6]
```diff
@@ -10,7 +10,8 @@ export enum RequestType {
     TRACK_SEARCH = "trackSearch",
     PLAYER_NAME = "playerName",
     SET_CONNECTION_ID = "setConId",
-    GET_VERSION = "version"
+    GET_VERSION = "version",
+    FCM_TOKEN = "token"
```
Comment (matthias-luger) [inline src/api/ApiTypes.d.tsx:14]: remove unused apitype

Comment (Ekwav) [reply src/api/ApiTypes.d.tsx:14]: how can it be used from the preScript?

Comment (matthias-luger) [reply src/api/ApiTypes.d.tsx:14]: it cant, the pre script is outside of typescript/react etc.

**Comment by Flou21** [comment]
ready to merge

File: src/utils/NotificationPermisson.tsx [hunk 7]
```diff
@@ -0,0 +1,13 @@
+export default async function askForNotificationPermissons() {
+    if (localStorage.getItem('fcmToken')) {
+        return;
+    }
+    // Retrieve an instance of Firebase Messaging so that it can handle background
+    // messages.
+    // @ts-ignore
+    let token = await window.messaging.getToken({
```
Comment (matthias-luger) [inline src/utils/NotificationPermisson.tsx:8]: remove await

**Comment by matthias-luger** [comment]
please fix all commented mistakes

File: src/components/Search/Search.tsx [hunk 8]
```diff
@@ -5,6 +5,9 @@ import './Search.css';
 import { useHistory } from "react-router-dom";
 import { convertTagToName } from '../../utils/Formatter';
 import NavBar from '../NavBar/NavBar';
+import { useMatomo } from '@datapunt/matomo-tracker-react';
```
Comment (matthias-luger) [inline src/components/Search/Search.tsx:8]: remove unused imports

**Comment by matthias-luger** [comment]
please fix commented mistakes/questions

File: public/preScript.js [hunk 9]
```diff
@@ -3,16 +3,22 @@ window.websocket.onopen = function() {
     console.log("Websocket opened");
 }
 
-if (window.location.href.toString().indexOf("localhost") !== -1) {
+if (window.location.href.toString().indexOf("localhost") !== -1 && false) {
```
Comment (matthias-luger) [inline public/preScript.js:6]: you really like this "&& false", do you?
Pls remove

Comment (Ekwav) [reply public/preScript.js:6]: the serviceworker is needed for push notifications, it should not be disabled on localhost like that

Comment (Ekwav) [reply public/preScript.js:6]: To Bypass the serviceworker cache check "DeveloperTools->Application->Service Workers->Bypass for network"

**Pull request merged**